content/beginner/194_secrets_manager/sync_native_secrets_env.md

```diff
@@ -0,0 +1,188 @@
+          key: db_username_01
+        - objectName: dbpassword
+          key: db_password_01
+EOF
+```
+
+Create custom resource.
+```bash
+kubectl apply -f nginx-deployment-spc-k8s-secrets.yaml
+
+kubectl get SecretProviderClass
+```
+
+The output indicates the resource ```nginx-deployment-spc-k8s-secrets``` created successfully.
+{{<output>}}
+NAME                               AGE
+nginx-deployment-spc               22m
+nginx-deployment-spc-k8s-secrets   1s
+{{</output>}}
+
+
+#### Create pod mount secrets volumes and set up Environment variables. 
+
+Configure a pod to mount volumes for individually extracted key-value pairs from the secrets. Once the pod is created with secrets volume mounts, the Secrets Store CSI Driver then creates and syncs Kubernetes secret object ```my-secret-01```. The pod then be able to populate Environment variables from the Kubernetes secret.
+
+```bash
+cat << EOF > nginx-deployment-k8s-secrets.yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment-k8s-secrets
+  labels:
+    app: nginx-k8s-secrets
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: nginx-k8s-secrets
+  template:
+    metadata:
+      labels:
+        app: nginx-k8s-secrets
+    spec:
+      serviceAccountName: nginx-deployment-sa
+      containers:
+      - name: nginx-deployment-k8s-secrets
+        image: nginx
+        imagePullPolicy: IfNotPresent
+        ports:
+          - containerPort: 80
+        volumeMounts:
+          - name: secrets-store-inline
+            mountPath: "/mnt/secrets"
+            readOnly: true
+        env:
+          - name: DB_USERNAME_01
+            valueFrom:
+              secretKeyRef:
+                name: my-secret-01
+                key: db_username_01
+          - name: DB_PASSWORD_01
+            valueFrom:
+              secretKeyRef:
+                name: my-secret-01
+                key: db_password_01
+      volumes:
+        - name: secrets-store-inline
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: nginx-deployment-spc-k8s-secrets
+EOF
+```
+
+Create the deployment and verify the creation of pods.
+```bash
+kubectl apply -f nginx-deployment-k8s-secrets.yaml
+sleep 2
+kubectl get pods -l "app=nginx-k8s-secrets"
+```
+
+#### Verify the result
+
+Verify the secret mounted as separate files for each extracted value and Environment variables. 
+
+```bash
+kubectl exec -it $(kubectl get pods | awk '/nginx-deployment-k8s-secrets/{print $1}' | head -1) -- /bin/sh
+```
+
+Wait for the shell prompt. Run the following set of commands and watch the output in the shell.  
+
+{{<output>}}
+# cd /mnt/secrets
+#
+# ls -l
+total 12
+-rw-r--r-- 1 root root 45 Oct  1 19:40 DBSecret_eksworkshop
+-rw-r--r-- 1 root root 12 Oct  1 19:40 dbpassword
+-rw-r--r-- 1 root root  3 Oct  1 19:40 dbusername
+# 
+# cat dbusername
+foo#
+#
+# cat dbpassword
```

Comment:
Change to `cat dbpassword ; echo` to add a newline after the print.

Reply:
Added trailing ‘;echo’ with several shell commands for cleaner output display

Changes applied with https://github.com/aws-samples/eks-workshop/pull/1271/commits/2fc29649acc6b75ff0b51f86291a41415b601998